packages/core/src/components/datetime/datetime.tsx

```diff
@@ -0,0 +1,244 @@
+    time: Intl.DateTimeFormatOptions
+  } = {
+    date: {
+      month: '2-digit',
+      day: '2-digit',
+      year: 'numeric',
+    },
+    time: {
+      hour: '2-digit',
+      minute: '2-digit',
+    },
+  }
+  @Prop() highlightedDates?: DatetimeHighlight[] | DatetimeHighlightCallback
+  @Prop() hourCycle: 'h12' | 'h23' = 'h23'
+  @Prop() hourValues?: number[] | string
+  @Prop() locale = 'pt-BR'
+  @Prop() isDateEnabled?: (dateString: string) => boolean
+  @Prop() max?: string
+  @Prop() min?: string
+  @Prop() minuteValues?: number[] | string
+  @Prop() monthValues?: number[] | string
+  @Prop() multiple?: boolean
+  @Prop() name?: string
+  @Prop() preferWheel = false
+  @Prop() presentation?: DatetimePresentation = 'date'
+  @Prop() rangeMode?: boolean
+  @Prop() readonly?: boolean
+  @Prop() showClearButton?: boolean
+  @Prop() showDefaultButtons?: boolean
+  @Prop() showDefaultTimeLabel = false
+  @Prop() showDefaultTitle = false
+  @Prop() useButton = false
+  @Prop() size?: 'cover' | 'fixed' = 'fixed'
+  @Prop({ mutable: true, reflect: true }) value?:
+    | DatetimeCustomEvent
+    | DatetimeChangeEventDetail
+  @Prop() yearValues?: number[] | string
+
+  @State() selectedDates: string[] = []
+
+  @Event() atomFocus!: EventEmitter<void>
+  @Event() atomBlur!: EventEmitter<void>
+  @Event() atomChange!: EventEmitter<string | string[]>
+  @Event() atomCancel!: EventEmitter<string>
+
+  private _datetimeEl!: HTMLIonDatetimeElement
+
+  get datetimeEl(): HTMLIonDatetimeElement {
+    return this._datetimeEl
+  }
+
+  set datetimeEl(value: HTMLIonDatetimeElement) {
+    this._datetimeEl = value
+  }
+
+  @Method()
+  async setValue(value) {
+    if (this.datetimeEl) {
+      this.datetimeEl.value = value
+    }
+  }
+
+  private readonly handleBlur = () => {
+    this.atomBlur.emit()
+  }
+
+  private readonly handleFocus = () => {
+    this.atomFocus.emit()
+  }
+
+  private readonly handleCancel = () => {
+    this.atomCancel.emit()
+  }
+
+  private calculateDateRange(dates: string[]): string[] {
+    const [start, end] = dates.map((date) => new Date(date))
+
+    const increment = start <= end ? 1 : -1
+    const daysDifference = Math.abs(
+      (end.getTime() - start.getTime()) / (1000 * 3600 * 24)
+    )
+
+    return Array.from({ length: daysDifference + 1 }, (_, i) => {
+      const currentDate = new Date(start)
+
+      currentDate.setDate(start.getDate() + i * increment)
+
+      return currentDate.toISOString().split('T')[0]
+    })
+  }
+
+  private handleRangeMode(dates: string[]) {
+    if (dates.length === 2) {
+      this.selectedDates = this.calculateDateRange(dates)
+    } else if (dates.length === 1) {
+      this.selectedDates =
+        this.selectedDates.length === 2
+          ? dates
+          : this.selectedDates.concat(dates)
+    } else {
+      this.selectedDates = []
+    }
+
+    this.atomChange.emit(this.selectedDates)
+  }
+
+  private handleDateChange = (event) => {
```

Comment:
I found this for typing, man :heart_eyes: 

```TS
import { JSX } from '@ionic/core/dist/types/components'

JSX.IonDatetime['onIonChange']
```

Reply:
Wow, I didn't know about it! Thanks a lot

[2cb1ff5](https://github.com/juntossomosmais/atomium/pull/687/commits/2cb1ff581c27a4532e8d2a0ab8f84178fa01f23a)